modules/vaos/spec/services/appointment_requests_service_spec.rb

```diff
@@ -5,13 +5,39 @@
 describe VAOS::AppointmentRequestsService do
   subject { described_class.for_user(user) }
 
-  let(:user) { build(:user, :mhv) }
-  let(:start_date) { Date.parse('2019-08-20') }
-  let(:end_date) { Date.parse('2020-08-22') }
+  let(:user) { build(:user, :vaos) }
 
   before { allow_any_instance_of(VAOS::UserService).to receive(:session).and_return('stubbed_token') }
 
+  describe '#put_cancel_appointment_request' do
+    context 'when appointment_request cannot be cancelled' do
+      let(:request_body)  { build(:va_appointment_request) }
+
+      it 'returns the bad request with detail in errors' do
+        VCR.use_cassette('vaos/appointment_requests/put_cancel_request_400', match_requests_on: %i[method uri]) do
+          expect { subject.put_cancel_appointment_request(request_body) }.to raise_error(
+            Common::Exceptions::BackendServiceException
+          )
+        end
+      end
+    end
+
+    context 'when appointment_request can be cancelled' do
+      let(:request_body)  { build(:va_appointment_request) }
+
+      it 'cancels the appointment_request' do
+        VCR.use_cassette('vaos/appointment_requests/put_cancel_request', match_requests_on: %i[method uri]) do
+          response = subject.put_cancel_appointment_request(request_body)
+          expect(response).to be_an_instance_of(String).and be_empty
+        end
+      end
+    end
```

Comment:
The request here is the same in both paths -- a new `build(:va_appointment_request)` but one results in an error and one doesn't. Seems like it would be clearer to show what changed to lead to a failure recording in particular. Or just pass some kind of mock request_body if the values aren't important.

The first spec says its a bad request but that's confusing when the second request, which seems exactly the same, is a valid request.

Reply:
I write these specs, and then I get on a staging box and I record the VCR cassettes. Then I commit the cassettes directly from staging machine. Perhaps you didn't notice that the requests identified are also not included as part of this PR. It's because they haven't been recorded yet. Which is why this is WIP.